File: src/ServerPulse.Backend/AnalyzerApi/Domain/Dtos/Responses/SlotDataResponse.cs
using AnalyzerApi.Domain.Dtos.Wrappers;
using AnalyzerApi.Domain.Models;

namespace AnalyzerApi.Domain.Dtos.Responses
{
    public class SlotDataResponse
    {
        public DateTime CollectedDateUTC { get; set; }
        public ServerStatistics? GeneralStatistics { get; set; }
        public ServerLoadStatistics? LoadStatistics { get; set; }
        public CustomEventStatistics? CustomEventStatistics { get; set; }
        public IEnumerable<LoadEventWrapper> LastLoadEvents { get; set; }
        public IEnumerable<CustomEventWrapper> LastCustomEvents { get; set; }
    }
}

File: src/ServerPulse.Frontend/src/app/modules/shared/domain/dtos/statistics/response/slotDataResponse.ts
import { CustomEventResponse, CustomEventStatisticsResponse, LoadEventResponse, ServerLoadStatisticsResponse, ServerStatisticsResponse } from "../../../../index";

export interface SlotDataResponse {
    collectedDateUTC: Date,
    generalStatistics: ServerStatisticsResponse | null,
    loadStatistics: ServerLoadStatisticsResponse | null,
    customEventStatistics: CustomEventStatisticsResponse | null,
    lastLoadEvents: LoadEventResponse[],
    lastCustomEvents: CustomEventResponse[],
}